rpcserver.go

```diff
@@ -517,6 +517,32 @@ func (r *rpcServer) MintAsset(ctx context.Context,
 		return nil, err
 	}
 
+	decDisplay := req.Asset.DecimalDisplay
+	metaType := req.Asset.AssetMeta.Type
+
+	// If decimal display is specified by the user, but the meta type is not
+	// JSON, then we have to error out as the value needs to be encoded in
+	// the meta field. If an opaque meta field is specified, then we simply
+	// don't know how to handle it.
+	if decDisplay != 0 && metaType != taprpc.AssetMetaType_META_TYPE_JSON {
+		return nil, fmt.Errorf("decimal display requires JSON meta " +
+			"type")
+	}
+
+	// If a JSON meta field is specified, then we'll update the decimal
+	// display in the meta field. Even if the decimal display is not
+	// specified in the rpc request, we'll still encode a 0 value.
+	if req.Asset.AssetMeta.Type == taprpc.AssetMetaType_META_TYPE_JSON {
+		updatedMeta, err := taprpc.EncodeDecimalDisplayInJSON(
```

Comment:
We should also cap the value of `decDisplay`. Given `uint64` is 19 decimal places, does it make sense to allow more than 12 to be after the decimal point?

Reply:
Agree

Reply:
Will add the above rationale as a comment next to the const